index.js

```diff
@@ -28,7 +28,7 @@ class PluginXrpPaychan extends PluginBtp {
     this._secret = opts.secret
     this._address = opts.address || deriveAddress(deriveKeypair(this._secret).publicKey)
 
-    this._currencyScale = opts.currencyScale || 6
+    this._currencyScale = (typeof opts.currencyScale === 'number') ? opts.currencyScale : 6
```

Comment:
Should this be an assertion instead and explicitly throw an error? Otherwise, it silently ignores non-number values.

`assert(typeof opts.currencyScale === 'number', 'currencyScale must be number')`

Reply:
Or, if currencyScale is a string, you could parse it to an int.

Reply:
should be `typeof opts.currencyScale === 'undefined' || typeof opts.currencyScale === 'number'`. I'll add that though. I don't think we should try to parse a string to a number; all other places where scale is used use numbers.